src/bors/handlers/retry.rs

```diff
@@ -0,0 +1,95 @@
+use std::sync::Arc;
+
+use crate::PgDbClient;
+use crate::bors::handlers::{PullRequestData, deny_request, has_permission};
+use crate::bors::merge_queue::MergeQueueSender;
+use crate::bors::{Comment, RepositoryState};
+use crate::github::{GithubUser, PullRequestNumber};
+use crate::permissions::PermissionType;
+
+pub(super) async fn command_retry(
+    repo_state: Arc<RepositoryState>,
+    db: Arc<PgDbClient>,
+    pr: PullRequestData<'_>,
+    author: &GithubUser,
+    merge_queue_tx: &MergeQueueSender,
+) -> anyhow::Result<()> {
+    if !has_permission(&repo_state, author, pr, PermissionType::Try).await? {
```

Comment:
I think that with the "new" implementation that is only for auto builds, it makes more sense to check the `review` permission instead. The set of people who have only the try permission and not the review permission is small anyway.